sled-agent/src/illumos/zone.rs

```diff
@@ -158,40 +158,44 @@ impl Zones {
                 };
 
                 if halt {
-                    zone::Adm::new(name).halt().map_err(|err| AdmError {
-                        op: Operation::Halt,
-                        zone: name.to_string(),
-                        err,
+                    zone::Adm::new(name).halt().await.map_err(|err| {
+                        AdmError {
+                            op: Operation::Halt,
+                            zone: name.to_string(),
+                            err,
+                        }
                     })?;
                 }
                 if uninstall {
-                    zone::Adm::new(name).uninstall(/* force= */ true).map_err(
-                        |err| AdmError {
+                    zone::Adm::new(name)
+                        .uninstall(/* force= */ true)
+                        .await
+                        .map_err(|err| AdmError {
                             op: Operation::Uninstall,
                             zone: name.to_string(),
                             err,
-                        },
-                    )?;
+                        })?;
                 }
                 zone::Config::new(name)
                     .delete(/* force= */ true)
                     .run()
+                    .await
                     .map_err(|err| AdmError {
-                        op: Operation::Delete,
-                        zone: name.to_string(),
-                        err,
-                    })?;
+                    op: Operation::Delete,
+                    zone: name.to_string(),
+                    err,
+                })?;
```

Comment:
what on earth is rustfmt doing here

Reply:
... uh... I think it's doing its best 😅

(I know this is weird, but I'm just gonna leave it, rather than try to poke the beast)